kernel/byterun/coq_interp.c

```diff
@@ -1771,6 +1773,63 @@ value coq_interprete
         Next;
       }
 
+
+      Instruct(ISINT_CAML_CALL2) {
+              print_instr("ISINT_CAML_CALL2");
+              if (Is_uint63(accu)) {
+                pc++;
+                print_int(*pc);
+                Setup_for_caml_call;
+                accu = caml_callback2(Field(coq_global_data, *pc),accu, sp[1]);
```

Comment:
That is a bit too frail. Please read `sp[0]` before `Setup_for_caml_call` rather than working under the assumption that the latter decreases the stack pointer by exactly one. (Same thing for the following new opcodes.)

Reply:
I'm more worried by the fact that `accu` is not protected. Is there a good reason not to protect it?

Note that there already are OCaml callbacks in `coq_uint63_emul.h`, it might be worth unifying this.
Lokking at this code, I note calls to `CAMLparam` and `CAMLreturn` macros. They might very well be "ceinture et bretelles" but are we sure?

Reply:
I had a look at OCAml's `callback.c` and it appears that the `CAMLparam` macros are indeed useless (they are already included in `caml_callback`), neither do we need to protect `accu`. So my previous comment can just be dismissed. Sorry for the noise.